graph/src/data/schema.rs

```diff
@@ -589,6 +594,33 @@ impl Schema {
         }
     }
 
+    /// Construct a value for the entity type's id attribute
+    pub fn id_value(&self, key: &EntityKey) -> Result<store::Value, Error> {
+        let base_type = self
+            .document
+            .get_object_type_definition(key.entity_type.as_str())
+            .ok_or_else(|| {
+                anyhow!(
+                    "Entity {}[{}]: unknown entity type `{}`",
+                    key.entity_type,
+                    key.entity_id,
+                    key.entity_type
+                )
+            })?
+            .field("id")
+            .unwrap()
+            .field_type
+            .get_base_type();
+
+        match base_type {
+            "String" => Ok(store::Value::String(key.entity_id.clone())),
+            "Bytes" => Ok(store::Value::Bytes(scalar::Bytes::from_str(
+                &key.entity_id,
+            )?)),
+            _ => todo!(),
```

Comment:
Could this `todo!` give a bad error message to an user (eg subgraph developer)?

Reply:
Ooops .. that should return an error; schema validation at some point should have checked that the `id` is either `String` or `Bytes`, but this shouldn't panic. I'll fix it.